### ISSUE TITLE
Can't get SD card to work with RCBugFix

### ISSUE THREAD

**Issue by funbobby2001** [MarlinFirmware/Marlin#4478]
-Ramps 1.4A; using motherboard 43
-Reprap discount full graphic smart controller
Worked fine in Marlin 1.0.2 which is the last version I had.
Controller recognizes that the card has been inserted but cannot see any of the files on the card.
Repetier host cannot see the files either, but Windows explorer can through my laptop card reader.

**Comment by Blue-Marlin** [comment]
#define SDSUPPORT

**Comment by funbobby2001** [comment]
It is

**Comment by Blue-Marlin** [comment]
Sorry i can't reproduce the problem.
Please try frormating - low level formatting the card.
If that does not help, please append the complete Configuration.h - renamed to *.txt.

**Comment by jbrazio** [comment]
Also check the card's filesystem, we only support `FAT`.

**Comment by funbobby2001** [comment]
It was FAT32. 
Low level formatting did not resolve the issue.

[Configuration.h.txt](https://github.com/MarlinFirmware/Marlin/files/394573/Configuration.h.txt)

**Comment by Blue-Marlin** [comment]
Your config. works with minimal changes in today's RCBugFix on my test system. (MANUAL_HOME_POSITIONS)

Check socket and card again.

**Comment by funbobby2001** [comment]
Just tried today's RCBugFix and that didn't work, so I guess it must be my board/socket.
The card reads/writes fine in Windows.

**Comment by thinkyhead** [comment]
@funbobby2001 If you revert back to 1.0.2 the SD card works okay?

**Comment by funbobby2001** [comment]
I know I should try that, just been dragging my feet because I have to uninstall Arduino and reinstall an older version to upload it.

**Comment by thinkyhead** [comment]
@funbobby2001 How odd. As far as I know older versions of Marlin should compile with the current (Arduino 1.6.x) toolkit.

**Comment by Blue-Marlin** [comment]
@thinkyhead 
There are several configs for 1.0.2 which do not compile with Arduino versions above 1.0.6.
Search for https://github.com/MarlinFirmware/Marlin/issues?q=is%3Aissue+POINTER_REGS+is%3Aclosed

**Comment by funbobby2001** [comment]
Now I can't get my old 1.0.2 to compile in Arduino 1.0.5

```
In file included from /dogm_lcd_implementation.h:38,
                 from ultralcd.cpp:39:
dogm_font_data_marlin.h:13: error: 'u8g_fntpgm_uint8_t' does not name a type
dogm_font_data_marlin.h:172: error: 'u8g_fntpgm_uint8_t' does not name a type
In file included from /dogm_lcd_implementation.h:40,
                 from ultralcd.cpp:39:
ultralcd_st7920_u8glib_rrd.h:43: error: 'u8g_t' was not declared in this scope
ultralcd_st7920_u8glib_rrd.h:43: error: 'u8g' was not declared in this scope
ultralcd_st7920_u8glib_rrd.h:43: error: 'u8g_dev_t' was not declared in this scope
ultralcd_st7920_u8glib_rrd.h:43: error: 'dev' was not declared in this scope
ultralcd_st7920_u8glib_rrd.h:43: error: expected primary-expression before 'msg'
ultralcd_st7920_u8glib_rrd.h:43: error: expected primary-expression before 'void'
ultralcd_st7920_u8glib_rrd.h:43: error: initializer expression list treated as compound expression
ultralcd_st7920_u8glib_rrd.h:44: error: expected ',' or ';' before '{' token
```

**Comment by Blue-Marlin** [comment]
Did you reinstall u8glib in a matching version? (https://github.com/MarlinFirmware/Marlin/tree/1.0.x/ArduinoAddons/Arduino_1.0.x/libraries/U8glib v1.1 then)

**Comment by funbobby2001** [comment]
Almost certainly not. LOL 
As soon as my print finishes I will try again. Thx

**Comment by funbobby2001** [comment]
Progress has been made.

1.0.2 now compiles and uploads with the older u8glib using Arduino 1.6.9, so that's nice
I've discovered that all the files on my SD card, which I've successfully printed from before, now have no file extension.

Why trying to upload files with Repetier Host the file extensions seem to be stripped.
If I manually copy files to the card with the file extension then everything is good.
How is it that the files on the card worked in the past and now they do not? Weird.
Either way I will not use the host to upload anymore because it takes FOREVER.
